### ISSUE TITLE
Error: type 'BoxValueConstraints<SliverStickyHeaderState?>' is not a subtype of type 'BoxValueConstraints<SliverStickyHeaderState>' in type cast

### ISSUE THREAD

**Issue by amitkot** [letsar/flutter_sticky_header#62]
Getting this error while using SliverStickyHeader:

```
flutter: type 'BoxValueConstraints<SliverStickyHeaderState?>' is not a subtype of type
flutter: 'BoxValueConstraints<SliverStickyHeaderState>' in type cast
flutter:
flutter: The relevant error-causing widget was:
flutter:   SliverStickyHeader
```

Using ```  flutter_sticky_header: ^0.6.0```.

More log info:
```
flutter: The following RenderObject was being processed when the exception was fired: _RenderValueLayoutBuilder<SliverStickyHeaderState>#4d71a relayoutBoundary=up4:
flutter:   creator: ValueLayoutBuilder<SliverStickyHeaderState> ← SliverStickyHeader ← ItemsAvailabilitySection
flutter:     ← MultiSliver ← ItemsByAvailability ← SliverOffstage ← Viewport ← IgnorePointer-[GlobalKey#d3e0b]
flutter:     ← Semantics ← Listener ← _GestureSemantics ←
flutter:     RawGestureDetector-[LabeledGlobalKey<RawGestureDetectorState>#aaeef] ← ⋯
flutter:   parentData: paintOffset=Offset(0.0, -0.0) (can use size)
flutter:   constraints: BoxConstraints(w=428.0, 0.0<=h<=Infinity)
flutter:   size: Size(428.0, 56.0)
flutter: This RenderObject had the following descendants (showing up to depth 5):
flutter:     child: RenderPadding#e0f35 relayoutBoundary=up5
flutter:       child: RenderParagraph#27526 relayoutBoundary=up6
flutter:         text: TextSpan
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
flutter: ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
flutter: The following _CastError was thrown during performLayout():
flutter: type 'BoxValueConstraints<SliverStickyHeaderState?>' is not a subtype of type
flutter: 'BoxValueConstraints<SliverStickyHeaderState>' in type cast
flutter:
flutter: The relevant error-causing widget was:
flutter:   SliverStickyHeader
flutter:   file:///xxx.dart:19:31
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #0      RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:1894:72)
flutter: #1      PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:915:15)
flutter: #2      RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:1894:14)
flutter: #3      RenderConstrainedLayoutBuilder.rebuildIfNecessary (package:flutter/src/widgets/layout_builder.dart:225:7)
flutter: #4      _RenderValueLayoutBuilder.performLayout (package:value_layout_builder/src/value_layout_builder.dart:101:5)
flutter: #5      RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
flutter: #6      RenderSliverStickyHeader.performLayout (package:flutter_sticky_header/src/rendering/sliver_sticky_header.dart:273:19)
flutter: #7      RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
flutter: #8      RenderMultiSliver.layoutChild (package:sliver_tools/src/rendering/multi_sliver.dart:293:13)
flutter: #9      RenderMultiSliver._layoutChildSequence (package:sliver_tools/src/rendering/multi_sliver.dart:155:31)
flutter: #10     RenderMultiSliver.performLayout (package:sliver_tools/src/rendering/multi_sliver.dart:90:24)
flutter: #11     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
flutter: #12     RenderSliverOffstage.performLayout (package:flutter/src/rendering/proxy_sliver.dart:316:12)
flutter: #13     RenderObject.layout (package:flutter/src/rendering/object.dart:1784:7)
flutter: #14     RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:512:13)
flutter: #15     RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1570:12)
flutter: #16     RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1479:20)
flutter: #17     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1641:7)
flutter: #18     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:884:18)
flutter: #19     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:453:19)
flutter: #20     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:874:13)
flutter: #21     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:319:5)
flutter: #22     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1144:15)
flutter: #23     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1082:9)
flutter: #24     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:998:5)
flutter: #28     _invoke (dart:ui/hooks.dart:161:10)
flutter: #29     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:253:5)
flutter: #30     _drawFrame (dart:ui/hooks.dart:120:31)
flutter: (elided 3 frames from dart:async)
```

**Comment by Peng-Qian** [comment]
please fix this type bug

**Comment by yasinarik** [comment]
@UnderKoen already made a PR for this package. I think it should be merged. Until then, if you want to fix this issue (and also want to use multiple slivers as a child):

1. Go to your project's `pubspec.yaml`
2. 
```
  # flutter_sticky_header: ^0.6.0 --> Commented this line
  # Added the below lines which basically points to UnderKoen's repo.
  flutter_sticky_header: 
  - git:
  - - url: https://github.com/UnderKoen/flutter_sticky_header
```

**Thanks a lot @UnderKoen!**

**Comment by letsar** [comment]
Fixed in 0.6.1, thank you @amitkot
